app/javascript/data.js

```diff
@@ -140,23 +149,56 @@ const validateDataset = () => {
   fetchDataMultiple(types, limits);
 };
 
-route('/data', () => {
+const themes = {
+  dark: 'ace/theme/monokai',
+  light: 'ace/theme/xcode'
+};
+let theme = localStorage.editorTheme || themes.dark;
+
+let editor;
+route('/data', async () => {
   preloadDataTypes();
   $('.schema-display').hide();
   $('.script-help').hide();
 
-  const editor = ace.edit('editor');
-  editor.setTheme('ace/theme/monokai');
+  await loadPromise($('.js-ace'), 'ace');
+  await loadPromise($('.js-jailed'), 'jailed');
+
+  const url = new URL(location.href);
+  url.pathname = '/data_sandbox.js';
+  const runner = new jailed.Plugin(url.toString());
+  runner.whenFailed(() => debug('failed to load runner'));
+
+  editor = ace.edit('editor');
+  if (localStorage.dataExplorerScriptContent) {
+    editor.getSession().getDocument().setValue(localStorage.dataExplorerScriptContent);
+  }
+  // Why? Because.
+  editor.$blockScrolling = Infinity;
   editor.getSession().setMode('ace/mode/javascript');
   editor.setOptions({
     minLines: 15,
     maxLines: 30,
     useSoftTabs: true,
-    tabSize: 4,
-    printMarginColumn: 120
+    tabSize: 2,
+    printMarginColumn: 80
```

Comment:
The JS code in the app uses 2 spaces. I’ll change the `printMarginColumn` part back.